### PR TITLE
Website finished

### PR DESCRIPTION
**Pull request by DOM-C1**

### CODE REVIEW

File: website/app.py [hunk 1]
```diff
@@ -1,30 +1,43 @@
...
+from utils_db import get_db_connection, get_id, setup_user_location, get_value_from_db
 
 
 app = Flask(__name__)
 
 
 @app.route('/')
 def homepage():
-    """This displays the homepage"""
+    """This displays the homepage."""
```
Comment (DKWeeto) [inline website/app.py:16]: Coool puncuation

File: website/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ 
+FROM python:latest
+
+copy . static
+COPY . app.py
+COPY . templates
+COPY . utils.py
+COPY . utils_db.py
+COPY .requirements.txt
```
Comment (DKWeeto) [inline website/Dockerfile]: This looks rushed should be "COPY [file_name] ."

File: website/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+ 
+FROM python:latest
+
+copy . static
```
Comment (DKWeeto) [inline website/Dockerfile]: capitalise copy

File: website/app.py [hunk 4]
```diff
@@ -33,19 +46,21 @@ def location_page():
...
+    name = request.form['name']
+    email = request.form['email']
+    sub_newsletter = request.form.get('newsletter', 'off') == 'on'
+    sub_alerts = request.form.get('alerts', 'off') == 'on'
+    details = get_details_from_post_code(location_value)
+    conn = get_db_connection()
+    if details['status'] == 200:
+        setup_user_location(details, name, email,
+                            sub_newsletter, sub_alerts, conn)
+    return render_template('cant_be_found_page.html')
```
Comment (DKWeeto) [inline website/app.py]: does this always return a cant_be_found_page? should this not be in an "else", with status 200 returning something else?

File: website/static/cant_be_found.jpg
Comment (DKWeeto) [inline website/static/cant_be_found.jpg:1]: yes!

File: website/static/cant_find_user.jpeg
Comment (DKWeeto) [inline website/static/cant_find_user.jpeg:1]: adorable!

File: website/static/thinking_cat.jpg
Comment (DKWeeto) [inline website/static/thinking_cat.jpg:1]: amazing!

File: website/utils_db.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+                         'county_id': county_id, 'longitude': longitude, 'latitude': latitude}
+        add_to_database('location', location_data, conn)
+        loc_id = get_loc_id(longitude, latitude, conn)
+
+    user_loc_data = {'user_id': user_id, 'loc_id': loc_id,
+                     'report_opt_in': sub_newsletter, 'alert_opt_in': sub_alerts}
+    add_to_database('user_location_assignment',
+                    user_loc_data, conn)
+    conn.close()
+    return
```
Comment (DKWeeto) [inline website/utils_db.py]: return wat?

File: website/utils_db.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+def setup_user_location(details, name, email, sub_newsletter, sub_alerts, conn) -> None:
+    """This sets up location tracking for a user, if the user exists then it just adds a new
+       location, otherwise, it sets up the new user too."""
+    longitude, latitude = get_long_lat(details)
+    location_name = get_location_name(details)
+    country = get_country(details)
+    county = get_county(details)
+    country_id = get_id('country', 'name', country, conn)
+    if country_id == -1:
+        return render_template('cant_be_found_page.html')
```
Comment (DKWeeto) [inline website/utils_db.py:66]: this return doesn't match the typehint deoes it?

File: website/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (DKWeeto) [inline website/Dockerfile]: Random blank space at start of file

**Review by DKWeeto — CHANGES_REQUESTED** [review]
Godd work generally, made some comments. Also maybe some big fucntions can be refactored into smaller ones?

File: website/Dockerfile (hunk 3 above)
Comment (DOM-C1) [reply website/Dockerfile]: amended

File: website/Dockerfile (hunk 7 above)
Comment (DOM-C1) [reply website/Dockerfile]: fixed

File: website/app.py (hunk 4 above)
Comment (DOM-C1) [reply website/app.py]: fixed

File: website/utils_db.py (hunk 5 above)
Comment (DOM-C1) [reply website/utils_db.py]: done

File: website/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,11 @@
+FROM python:latest
+
+WORKDIR /app
+
+COPY . /app
```
Comment (DKWeeto) [inline website/Dockerfile:5]: i believe this is bad practice to COPY everything

**Review by DKWeeto — APPROVED** [review]

**Pull request merged**